pennylane/resource/specs.py

```diff
@@ -21,18 +22,43 @@ def _get_absolute_import_path(fn):
     return f"{inspect.getmodule(fn).__name__}.{fn.__name__}"
 
 
-def specs(qnode, max_expansion=None, expansion_strategy=None):
-    """Resource information about a quantum circuit.
+def _determine_spec_level(kwargs, qnode):
+    if "max_expansion" in kwargs:
+        warnings.warn(
+            "'max_expansion' has no effect on the output of 'specs()' and should not be used."
+        )
+
+    sentinel = object()
+
+    level = kwargs.get("level", sentinel)
+    expansion_strategy = kwargs.get("expansion_strategy", sentinel)
+
+    if all(val != sentinel for val in (level, expansion_strategy)):
+        raise ValueError("Either 'level' or 'expansion_strategy' need to be set, but not both.")
+
+    if level == sentinel and expansion_strategy == sentinel:
+        return qnode.expansion_strategy
+
+    if level == sentinel:
+        return expansion_strategy
+
+    return level
+
+
+def specs(qnode, **kwargs):
+    r"""Resource information about a quantum circuit.
 
     This transform converts a QNode into a callable that provides resource information
-    about the circuit.
+    about the circuit after applying the specified amount of transforms/expansions first.
 
     Args:
         qnode (.QNode): the QNode to calculate the specifications for
 
     Keyword Args:
-        max_expansion (int): The number of times the internal circuit should be expanded when
-            calculating the specification. Defaults to ``qnode.max_expansion``.
+        level (None, str, int, slice): An indication of what transforms to apply before computing the resource information.
+            Check :func:`~.workflow.get_transform_program` for more information on the allowed values and usage details of
```

Comment:
Note that there is a typo in `get_transform_program`:
> And indication of what transforms to use from the full program.

First word shouldn't be And